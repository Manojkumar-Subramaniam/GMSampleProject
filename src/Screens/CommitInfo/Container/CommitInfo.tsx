import React from 'react'
import { View, Text } from 'react-native'
import { useRoute } from '@react-navigation/native';
import styles from '../Styles'
import { Strings, Fonts } from '../../../Utils'

const CommitInfo: React.FC = () => {

    const route = useRoute()
    const { data } = route.params

    return (
        <View style={styles.rootView}>
            <CommitItem field={Strings.COMMIT_HASH} value={data.sha} />
            <CommitItem field={Strings.COMMIT_AUTHOR} value={data.commit.author.name} />
            <CommitItem field={Strings.COMMIT_MESSAGE} value={data.commit.message} />
        </View>
    )
}

interface commitItemProps {
    field: string
    value: string
}

const CommitItem: React.FC<commitItemProps> = (props) => {
    return (
        <Text style={[Fonts.fontNormalBlack, { flexDirection: 'row' }]}>
            <Text style={Fonts.fontNormalHighlighted} >{props.field}</Text>
            <Text>{Strings.COLON}</Text>
            <Text>{props.value}</Text>
        </Text>
    )
}

export default CommitInfo