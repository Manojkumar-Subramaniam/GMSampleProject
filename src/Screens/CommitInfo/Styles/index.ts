import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  rootView: {
    margin: 16,
  },
});

export default styles;
