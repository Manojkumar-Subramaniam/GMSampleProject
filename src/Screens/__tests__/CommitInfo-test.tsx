import React from 'react';
import renderer from 'react-test-renderer';
import CommitInfo from '../CommitInfo/Container/CommitInfo';
import gitCommitAPI from '../../../mockStore';
import {useRoute} from '@react-navigation/native';
jest.mock('@react-navigation/native');

describe('GM Home', () => {
  useRoute.mockReturnValue({
    params: {data: gitCommitAPI[0]},
  });

  test('Commit Info', () => {
    const commitInfoSnapshot = renderer.create(<CommitInfo />).toJSON();
    expect(commitInfoSnapshot).toMatchSnapshot();
  });
});
