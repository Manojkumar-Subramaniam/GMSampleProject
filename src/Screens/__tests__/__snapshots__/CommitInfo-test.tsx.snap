// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GM Home Commit Info 1`] = `
<View
  style={
    Object {
      "margin": 16,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "normal",
        },
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#34a8eb",
          "fontSize": 16,
          "fontWeight": "normal",
        }
      }
    >
      Hash
    </Text>
    <Text>
      : 
    </Text>
    <Text>
      e29a7e3074e477d9567aa0953f5c6171ba805e1c
    </Text>
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "normal",
        },
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#34a8eb",
          "fontSize": 16,
          "fontWeight": "normal",
        }
      }
    >
      Author
    </Text>
    <Text>
      : 
    </Text>
    <Text>
      Manojkumar Subramaniam
    </Text>
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "normal",
        },
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#34a8eb",
          "fontSize": 16,
          "fontWeight": "normal",
        }
      }
    >
      Message
    </Text>
    <Text>
      : 
    </Text>
    <Text>
      created RN Sample project
    </Text>
  </Text>
</View>
`;
